### ISSUE TITLE
SLF001 when using Django Model `_meta` API

### ISSUE THREAD

**Issue by bittner** [astral-sh/ruff#13382]
In Django, there is the [Model `_meta` API](https://docs.djangoproject.com/en/stable/ref/models/meta/), an endorsed way to access internal information about a model.

Example model:

```python
class MyModel(models.Model):

    class Meta:
        db_table = "custom_table_name"
```

Example usage:

```python
table_name = MyModel()._meta.db_table
```

The way the API is used is by the `_meta` member, which quite obviously triggers [SLF001](https://docs.astral.sh/ruff/rules/private-member-access/) as a linter warning. (As of Ruff version **0.6.4**)

There's no other way to access the members of this API, and as this is an official API of the Django framework it's not clear why code like this should be discouraged (unless it's about motivating the Django project to redesign this API).

Is there any chance we could make Ruff consider this as an accepted pattern? -- Should this request be submitted to the [flake8-django](https://github.com/rocioar/flake8-django) project, first?

**Comment by charliermarsh** [comment]
I think I'd recommend adding this to [`ignore-names`](https://docs.astral.sh/ruff/settings/#lint_flake8-self_ignore-names) -- does that work?

E.g.:

```toml
[tool.ruff.lint.flake8-self]
ignore-names = ["_meta"]
```

**Comment by bittner** [comment]
That works, thank you! :+1: 

But shouldn't we make this a general rule, and more specific, just for Django Model classes?

**Comment by MichaReiser** [comment]
> But shouldn't we make this a general rule, and more specific, just for Django Model classes?

That would be great, but Ruff must understand that `models.Model` inherits a Django model, something it currently doesn't understand because it lacks multifile analysis. 

An alternative is to default to ignoring `_meta` whenever there's a `django` import, but I don't even know if that's the case for this file. This has the risk of the rule missing `_meta` usages outside of a Django model. 

Overall, ignore names is a good explicit opt in for now.
